src/installer/managed/Microsoft.NET.HostModel/Bundle/Manifest.cs

```diff
@@ -133,8 +133,9 @@ private string GenerateDeterministicId()
             byte[] manifestHash = bundleHash.Hash;
             bundleHash.Dispose();
             bundleHash = null;
-
-            return Convert.ToBase64String(manifestHash).Substring(BundleIdLength).Replace('/', '_');
+            string id = Convert.ToBase64String(manifestHash).Substring(0, BundleIdLength).Replace('/', '_');
```

Comment:
More safer escaping:

```c#
using using System.Buffers.Text;
...
#if NET
            string id = Base64Url.EncodeToString(manifestHash).Substring(0, BundleIdLength);
#else
            string id = Convert.ToBase64String(manifestHash).Substring(0, BundleIdLength).Replace('/', '_');
#endif
```
If without `#if / #else` directives is desired, `<PackageReference Include="System.Buffers" Version="$(SystemBuffersVersion)" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />`

Reply:
Are these guaranteed to be identical? I'd be worried about using a different API in different TFMs when the id needs to be deterministic.

Also, what do you mean by safer?

This was written this way because Roslyn does this for getting a unique pipe name and hasn't had issues with it: https://github.com/dotnet/runtime/pull/52930#discussion_r635538722.
Current implementation:
https://github.com/dotnet/roslyn/blob/c8b5f306d86bc04c59a413ad17b6152663a1e744/src/Compilers/Shared/BuildServerConnection.cs#L579

Reply:
They are not identical but it would be more correct usage anyway.

`Base64Url.EncodeToString` differs in three ways:
- Replaces `/` => `_` (same as in previous code)
- Removes padding with `=` at the end (irrelevant in this case because we trim at 12 characters to the padding is never part of the output)
- **Replaces `+` => `-`**

Reply:
It looks like Base64Url is not in the maintainence packages, but I can add a separate implementation for Framework.